src/wallet/test/walletdb_tests.cpp

```diff
@@ -48,10 +48,13 @@ BOOST_AUTO_TEST_CASE(walletdb_read_write_deadlock)
         LOCK(wallet->cs_wallet);
         auto legacy_spkm = wallet->GetOrCreateLegacyScriptPubKeyMan();
         BOOST_CHECK(legacy_spkm->SetupGeneration(true));
+        const auto& db_batch = wallet->GetDatabase().MakeBatch();
+        BOOST_CHECK(HasLegacyRecords(wallet.get(), *db_batch));
```

Comment:
This leaves the db batch object alive and could conflict with the `Flush()` and `DeleteRecords()` internals, as both access the db.

Better to limit the scope of the db batch object only to this function call.